src/components/LocationsTable.vue

```diff
@@ -0,0 +1,121 @@
+<template>
+  <section class="locations-table">
+    <div class="locations-table__filters-wrapper">
+      <SearchInput
+        v-for="({ ariaLabel, paramKey, placeholder, items, type }, index) in locationsFilters"
+        class="locations-table__filter"
+        :key="`${index}${paramKey}`"
+        :type="type"
+        :placeholder="placeholder"
+        :ariaLabel="ariaLabel"
+        :items="items"
+        :selectedValue="searchParams[paramKey]"
+        @filterChange="(value) => setSearchParams({ [paramKey]: value })"
+      />
+    </div>
+    <div class="locations-table__grid-wrapper">
+      <AgGridVue
+        class="locations-table__grid"
+        :theme="gridTheme"
+        :loading="locationsStore.isFetchingLocations"
+        :defaultColDef="defaultLocationsColDef"
+        :columnDefs="locationsColDefs"
+        :rowData="locationsStore.locations"
+        @gridReady="onGridReady"
+      />
+      <div class="locations-table__pagination-wrapper">
+        <Pagination
+          class="locations-table__pagination"
+          :currentPage="searchParams.page"
+          :totalPagesNumber="locationsStore.pages"
+          @pageChange="(page) => setSearchParams({ page })"
+        />
+      </div>
+    </div>
+  </section>
+</template>
+
+<script setup lang="ts">
+import { shallowRef, watch } from 'vue'
+import { AgGridVue } from 'ag-grid-vue3'
+import type { GridApi, GridReadyEvent } from 'ag-grid-community'
+
+import { useSearchParams } from '@/composables/useSearchParams'
+import { gridTheme } from '@/theme'
+import type { Location } from '@/services/locations'
+import { defaultLocationsColDef, locationsColDefs } from '@/helpers/columnDefinitions'
+import { useLocationsStore } from '@/stores/locationsStore'
+import { locationsFilters } from '@/constants'
+
+import Pagination from './Pagination.vue'
+import SearchInput from './SearchInput.vue'
+
+export type LocationsTableSearchParams = {
+  page: number
+  name: string
+  type: string
+  dimension: string
+}
+
+const gridApi = shallowRef<GridApi<Location> | null>(null)
+
+const locationsStore = useLocationsStore()
+const { searchParams, setSearchParams } = useSearchParams<LocationsTableSearchParams>()
+
+const onGridReady = (params: GridReadyEvent) => {
+  gridApi.value = params.api
+
+  locationsStore.fetchLocations(searchParams.value)
+}
+
+watch(searchParams, () => locationsStore.fetchLocations(searchParams.value))
+</script>
+
+<style scoped lang="scss">
+.locations-table {
+  $table-height: 488px;
+  $max-width-table: 964px;
+
+  display: flex;
+  flex-direction: column;
+  gap: $space-xs;
+  max-width: $max-width-table;
+  margin: 0 auto;
```

Comment:
it looks identical like .episodes-table. Maybe let's use some global class or create a common wrapper for this component smth like typography